libraries/chain/include/eosio/chain/webassembly/eos-vm-oc/code_cache.hpp

```diff
@@ -80,7 +80,7 @@ class code_cache_base {
 
       //these are really only useful to the async code cache, but keep them here so
       //free_code can be shared
-      std::unordered_set<code_tuple> _queued_compiles;
+      deque<code_tuple> _queued_compiles;
```

Comment:
I'm worried about the performance ramification of switching this from constant time look up to linear time lookup. The lookup occurs every time an action is executed and when someone starts fresh there could be hundreds or thousands of entries that need to be searched through for every action (until compilation settles down).

I think maybe instead we should have a second
```c++
std::unordered_set<code_tuple> _high_priority_queued_compiles;
```

Reply:
One issue I saw was that compiling was according to code_hash sort order instead of order they came in. This preserves the order. If worried about performance, maybe we should use a multindex instead for this which would allow for keeping track of order, last used, prioritize eosio.*, and provide fast lookup.

Reply:
yeah if you want to maintain compilation order, a multi_index sounds good

Reply:
Done